src/main/groovy/com/rundeck/plugins/ansible/ansible/AnsibleInventoryList.java

```diff
@@ -0,0 +1,84 @@
+package com.rundeck.plugins.ansible.ansible;
+
+import com.rundeck.plugins.ansible.util.ProcessExecutor;
+import lombok.Builder;
+import lombok.Data;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@Builder
+public class AnsibleInventoryList {
+
+    private String inventory;
+    private String configFile;
+    private boolean debug;
+
+    public static final String ANSIBLE_INVENTORY = "ansible-inventory";
+
+    /**
+     * Executes Ansible command to bring all nodes from inventory
+     * @return output in yaml format
+     */
+    public String getNodeList() {
+
+        List<String> procArgs = new ArrayList<>();
+        procArgs.add(ANSIBLE_INVENTORY);
+        procArgs.add("--inventory-file" + "=" + inventory);
+        procArgs.add("--list");
```

Comment:
There are two things we need to include here from the plugin
1) if the field 'limit' is set, it should be added an argument to the command: eg:
2) if the inventory file is encrypted, we need to pass the vault-password

Reply:
Done... ✅